static/js/main.js

```diff
@@ -0,0 +1,252 @@
+/**/
+(function($) {
```

Comment:
Please split the externals JavaScript frameworks into several functional commits. Also, it looks file `static/js/main.js` was written by you, right? If so, please decouple from this commit because we always want to ease the maintenance of external components.